frontend/src/pages/projects/components/DeleteModal.tsx

```diff
@@ -34,56 +34,66 @@ const DeleteModal: React.FC<DeleteModalProps> = ({
     }
   };
 
+  const handleSubmit = (e: React.KeyboardEvent<object>) => {
+    if (e.key === 'Enter' && value !== deleteName) {
+      e.preventDefault();
+    } else if (e.key === 'Enter' && value === deleteName) {
+      onDelete();
+    }
+  };
+
   React.useEffect(() => {
     if (!isOpen) {
       setValue('');
     }
   }, [isOpen]);
 
   return (
-    <Modal
-      title={title}
-      titleIconVariant="warning"
-      isOpen={isOpen}
-      onClose={() => onBeforeClose(false)}
-      actions={[
-        <Button
-          key="delete-button"
-          variant="danger"
-          isLoading={deleting}
-          isDisabled={deleting || value !== deleteName}
-          onClick={() => onBeforeClose(true)}
-        >
-          {submitButtonLabel}
-        </Button>,
-        <Button key="cancel-button" variant="secondary" onClick={() => onBeforeClose(false)}>
-          Cancel
-        </Button>,
-      ]}
-      variant="small"
-    >
-      <Stack hasGutter>
-        <StackItem>{children}</StackItem>
-        <StackItem>
-          Confirm deletion by typing <strong>{deleteName}</strong> below:
-        </StackItem>
-        <StackItem>
-          <TextInput
-            id="delete-modal-input"
-            aria-label="Delete modal input"
-            value={value}
-            onChange={(newValue) => setValue(newValue)}
-          />
-        </StackItem>
-        {error && (
+    <Form onKeyDown={handleSubmit}>
```

Comment:
Form should go into the body of the modal... I'm fairly certain this modal will create a portal 🤔

Reply:
Also -- if the solution is to use a key listener... perhaps we just add that to the text field itself? (and skip the Form all together)

Reply:
I agree on this for the Delete modal but not sure if this can be applied to other modals which include "Description" fields(coz Description fields may include newlines. Apologies if this is out of context. Just stating it here to keep in mind when we are dealing with other modals' accessibility.

Reply:
> Form should go into the body of the modal... I'm fairly certain this modal will create a portal 🤔

Thanks @andrewballantyne for quick review. Actually the idea was to prevent closing the modal initially when user hit the enter key but yes, as you suggested it can be done using textfield itself. I will be updating accordingly.

Reply:
Updated the PR according. Thanks

Reply:
Looks good 👍